runtime/jcl/common/jcldefine.c

```diff
@@ -165,8 +166,7 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 				 */
 #if defined(J9VM_OPT_SNAPSHOTS)
 				if (IS_RESTORE_RUN(vm)) {
-					clazz = vmFuncs->hashClassTableAt(classLoader, utf8Name, utf8Length, 0);
-
+					clazz->classFlags &= ~J9ClassIsJCLDefineClass;
```

Comment:
Strictly speaking, this is not needed. We may even want to keep the flag to support things like multiple snapshots.